configs/builder_type_cfg.yaml

```diff
@@ -18,13 +18,32 @@ builders:
     commands:
       - command: npm install --production
 
+  - name: npm-ci
+    info: "clean install production dependencies"
+    path: "path to config file which override the following default commands"
+    commands:
+      - command: npm clean-install --production
+
+  - name: npm-ci-dev
```

Comment:
Thanks for your contribution!  

`npm-ci` looks good to me and commonly used, however, `grunt-ci` and `npm-ci-dev` should be handled as a custom builder as they are less common for production use and we want to avoid pollution of builders.
in addition, as this config is transferred to binary, please run the following command on the root to update the builder binary fiile 

`//go:generate go run ./internal/buildtools/embed.go ./internal/buildtools/buildtools_msg.go -source=./configs/builder_type_cfg.yaml -target=./internal/commands/builder_type_cfg.go -name=BuilderTypeConfig -package=commands
`
if you have an issue please let me know

Reply:
I included `npm-ci-dev` only because you already had it. I agree that this is one is mostly nonsense in production. `grunt-ci` on the other hand is the only sane way to run grunt in my opinion. The problem with not using clean install is that you introduce changes completely outside of your control into your pipeline.
Either way, I can run custom builders so that's not a request just my opinion.

I will run the update on Monday